app/main.py

```diff
@@ -2,5 +2,19 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    def inner(*args) -> int:
+
+        for item in cash:
+            if item[0] == func and item[1] == args:
+                print("Getting from cache")
+                return item[2]
+
+        print("Calculating new result")
+        result = func(*args)
+        cash.append([func, args, result])
+        return result
+
+    return inner
+
+
+cash = [[None, None, None]]
```

Comment:
cache  is a global variable, which is not a good practice as it can lead to conflicts and unintended side-effects. It is better to encapsulate the cache within the decorator itself.

Reply:
cache is implemented as a list of lists, which is inefficient for lookups. A dictionary would be more appropriate